### PR TITLE
Install Pluto.jl and jupyter-pluto-proxy

### PR DESCRIPTION
**Pull request by yuvipanda**
## Describe your changes

[Pluto.jl](https://plutojl.org/) is an alternative reactive notebook frontend focused specifically on Julia. I think shipping this by default in the julia-enabled images helps serve the Julia community better, particularly when used with JupyterHub.

For context, I am working with the Julia users of the [Jupyter Meets the Earth](https://jupytearth.org/) project, and trying to understand how to best serve their needs on a JupyterHub. We currently maintain a massive image that 'has everything', but I'm trying to instead work upstream wherever possible so everyone working in these subfields can benefit. Meeting Julia users where they are at seems a useful path forward here.

## Checklist (especially for first-time contributors)

- [x] I have performed a self-review of my code
- [ ] If it is a core feature, I have added thorough tests
- [x] I will try not to use force-push to make the review process easier for reviewers
- [x] I have updated the documentation for significant changes

### CODE REVIEW

**Comment by yuvipanda** [comment]
If this looks acceptable, I'll write a specific test for this too.

File: docs/using/selecting.md [hunk 1]
```diff
@@ -118,6 +118,7 @@ It contains:
 - Everything in `jupyter/minimal-notebook` and its ancestor images
 - The [Julia Programming Language](https://julialang.org/)
 - [IJulia](https://github.com/JuliaLang/IJulia.jl) to support Julia code in Jupyter notebook
+- [Pluto.jl](https://plutojl.org/) reactive Julia notebook interface, made accessible with [jupyter-pluto-proxy](https://github.com/yuvipanda/jupyter-pluto-proxy)
```
Comment (mathbunnyru) [inline docs/using/selecting.md:121]: This has to be added to 2 places, unfortunately.

File: tests/base-notebook/test_packages.py [hunk 2]
```diff
@@ -93,7 +93,7 @@ def packages(package_helper: CondaPackageHelper) -> dict[str, set[str]]:
 
 def get_package_import_name(package: str) -> str:
     """Perform a mapping between the python package name and the name used for the import"""
-    return PACKAGE_MAPPING.get(package, package)
+    return PACKAGE_MAPPING.get(package, package).replace("-", "_")
```
Comment (mathbunnyru) [inline tests/base-notebook/test_packages.py:96]: Let's just add one more package to PACKAGE_MAPPING?

**Review by mathbunnyru — COMMENTED** [review]
To be honest, I know nothing about Julia or Julia's infrastructure.
So, @yuvipanda, if you know someone from Julia (+Jupyter) community, who can take a look here and approve this change, this would be awesome.
As I understand - do we need to run something special to run Pluto?
If yes, maybe we should probably create a recipe with how to do that.
https://jupyter-docker-stacks.readthedocs.io/en/latest/contributing/recipes.html

File: tests/base-notebook/test_packages.py (hunk 2 above)
Comment (yuvipanda) [reply tests/base-notebook/test_packages.py]: Done!

File: docs/using/selecting.md (hunk 1 above)
Comment (yuvipanda) [reply docs/using/selecting.md:121]: Done!

**Comment by yuvipanda** [comment]
@mathbunnyru you don't need anything special to use it. You can see it in the JupyterLab launcher!

image

So I don't think a specific recipe is needed.

**Comment by yuvipanda** [comment]
I'm going to add a simple test to make sure this works.

**Comment by yuvipanda** [comment]
ok, I added a test to make sure this starts! I've also asked some julia users to try comment on this, just as social proof :) 

Anything else you'd like me to do, @mathbunnyru?

**Review by mathbunnyru — APPROVED** [review]
Thank you for adding the test.
Let's use the DRY principle and move the test code to a function in `tests/julia-tests/pluto_proxy.py` folder and in the `julia-notebook` and `datascience-notebook` folders add a test which will be calling this function.

**Comment by JordiBolibar** [comment]
Thanks a lot for all the efforts on building a Julia-specific image for the hub! 

I've been heavily using the "massive" image from Jupyter Meets the Earth the last 2.5 years, and I can say that adding Pluto.jl would be fantastic. Having Julia + Pluto.jl + VSCode + Jupyter notebooks with CPUs is pretty much everything we could ask for 🙂

**Comment by benz0li** [comment]
@JordiBolibar There is also the [JupyterLab Julia docker stack](https://github.com/b-data/jupyterlab-julia-docker-stack) and the [Data Science Development Containers](https://github.com/b-data/data-science-devcontainers).  
ℹ️ (GPU accelerated) Multi-arch (`linux/amd64`, `linux/arm64/v8`) container images

**Comment by mathbunnyru** [comment]
> Thanks a lot for all the efforts on building a Julia-specific image for the hub!
> 
> I've been heavily using the "massive" image from Jupyter Meets the Earth the last 2.5 years, and I can say that adding Pluto.jl would be fantastic. Having Julia + Pluto.jl + VSCode + Jupyter notebooks with CPUs is pretty much everything we could ask for 🙂

Great 👍
It always feels great to make users of our images happy 🙂

**Comment by mathbunnyru** [comment]
I think I'll just merge this and then remove code duplication in the `main` branch.
I want to put all the images to the `images` folder and this is blocking me a little bit.

**Comment by yuvipanda** [comment]
Thank you so much, @mathbunnyru! And thanks for chiming in, @JordiBolibar!

**Pull request merged**